### PR TITLE
Add support for sharp 0.30

### PR DESCRIPTION
**Pull request by rexxars**
(Closes #58583 as these typings are stricter/more correct)

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://sharp.pixelplumbing.com/changelog#v030---dresser
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/sharp/index.d.ts [hunk 1]
```diff
@@ -727,6 +733,8 @@ declare namespace sharp {
          * An integral Number of pixels, zero or false to remove limit, true to use default limit of 268402689 (0x3FFF x 0x3FFF). (optional, default 268402689)
          */
         limitInputPixels?: number | boolean | undefined;
+        /** Set this to true to remove safety features that help prevent memory exhaustion (SVG, PNG). (optional, default false) */
+        unlimited?: boolean | undefined;
```
Comment (fdebijl) [inline types/sharp/index.d.ts:737]: Optional params are union typed with `undefined` implictly \[1], so we could drop the `|  undefined` on all the optional params I think.

\[1]: https://www.typescriptlang.org/docs/handbook/2/functions.html#optional-parameters

**Review by fdebijl — APPROVED** [review]
LGTM, thanks for the thorough update! If you want you can drop the `undefined` unions, but since previous params also have the explicit `undefined` it's not a dealbreaker for me.

File: types/sharp/index.d.ts (hunk 1 above)
Comment (rexxars) [reply types/sharp/index.d.ts:737]: Yeah, I figured I would just keep things consistent with the current style, but will consider doing another PR to remove the superfluous `| undefined`  on all of these

**Comment by rexxars** [comment]
Ready to merge

**Pull request merged**